src/server.py

```diff
@@ -27,38 +31,53 @@ def server():
             if len(part) < buffer_length:
                 message_complete = True
         print(message)
-        conn.send(response_ok())
+        try:
+            parse_request(message)
+        except HTTPException as exception:
+            conn.send((response_error(exception).encode("utf-8")))
+            conn.close()
+        else:
+            conn.send(response_ok())
```

Comment:
One other thing.  Remember that `send` returns an integer value representing the number of bytes successfully sent.  It is your job if you use it to check that number against the length of the message you tried to send to be sure that it all went.

If you use `socket.sendall(msg)` instead, you can be sure that if the method returns without error, the whole message was sent.